### PR TITLE
Added exporter.js to clean up importing in app and components

### PR DESCRIPTION
**Pull request by Johnr302**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
 import "bootstrap/dist/css/bootstrap.min.css";
-import { PLAYER } from "./constants";
 
 function App() {
   const [player, setPlayer] = useState("X");
   const [gameOver, setGameOver] = useState(false);
+  const [result, setResult] = useState("");
+  const [board, setBoard] = useState(BOARD_STATE);
+
+  const updateBoard = (event, key, player) => {
```
Comment (d48) [inline src/App.js]: Is this being used yet?

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
-import EndGameDeclaration from "./components/EndGameDeclaration";
-import NewGame from "./components/NewGame";
-import Board from "./components/Board";
+import {
+  PlayerTurn,
+  EndGameDeclaration,
+  NewGame,
+  Board,
+  winLogic,
+} from "./exporter";
```
Comment (d48) [inline src/App.js:9]: Nice. Some people just reference path to "components/index" and index is that export file that is a manifest of all components. Good find.

File: src/test/App.test.js [hunk 3]
```diff
@@ -36,7 +39,19 @@ describe("App", () => {
     props = playerTurn.props();
     expect(props.player).toBe(PLAYER.X);
   });
-  it("can determine X as winner", () => {});
+  it("can determine X as winner", () => {
```
Comment (d48) [inline src/test/App.test.js:43]: Does this work with the updates or not yet?

Comment (Johnr302) [reply src/test/App.test.js:43]: It  does not.

File: src/App.js (hunk 1 above)
Comment (Johnr302) [reply src/App.js]: This has not been implemented.

Comment (d48) [reply src/App.js]: ok cool

File: src/test/App.test.js (hunk 3 above)
Comment (d48) [reply src/test/App.test.js:43]: ok

**Comment by d48** [comment]
good to merge

File: src/components/Board.js [hunk 4]
```diff
@@ -2,18 +2,21 @@ import React from "react";
...
-  }
-  return <section className="board">{result}</section>;
+  const { setPlayerClickHandler, gameOver, currentPlayer, board } = props;
+  return (
+    <section id="board" className="board">
+      {board.map((board, index) => {
+        return (
+          <Square
+            key={index}
+            id={board.id}
```
Comment (d48) [inline src/components/Board.js:12]: I didn't notice the object having an id. Is it present?

File: src/App.js [hunk 5]
```diff
@@ -1,36 +1,88 @@
...
 
-  const setPlayerClickHandler = () => {
-    setPlayer((currentPlayer) => {
-      if (currentPlayer === PLAYER.X) {
-        return PLAYER.O;
-      }
-      return PLAYER.X;
+  const newGameClickHandler = () => {
+    setBoard(BOARD_STATE);
+    setGameOver(false);
```
Comment (d48) [inline src/App.js:21]: Nice

File: src/App.js [hunk 6]
```diff
@@ -1,36 +1,88 @@
...
+    const marked = (currentValue) => currentValue.marked !== "";
+    return board.every(marked);
+  };
+
+  useEffect(() => {
+    const isWinner = winLogic(board, player);
+    if (isWinner) {
+      setGameOver(true);
+      setResult(player);
+    } else if (isBoardFull(board)) {
```
Comment (d48) [inline src/App.js:54]: Very nice

File: src/test/App.test.js [hunk 7]
```diff
@@ -2,28 +2,33 @@ import React from "react";
...
+  PlayerTurn,
+  Board,
+  EndGameDeclaration,
+  Square,
+  NewGame,
+} from "../exporter";
 import { PLAYER } from "../constants";
 
-Enzyme.configure({ adapter: new Adapter() });
+//To do create components index for cleaner import
```
Comment (d48) [inline src/test/App.test.js]: Remove Comment

File: src/test/App.test.js [hunk 8]
```diff
@@ -36,9 +41,68 @@ describe("App", () => {
...
+    square.at(3).simulate("click"); //O
+    square.at(5).simulate("click"); //X
+    square.at(4).simulate("click"); //0
+    square.at(6).simulate("click"); //X
+    square.at(8).simulate("click"); //0
+    square.at(7).simulate("click"); //X
+    const endGame = wrapper.find(EndGameDeclaration);
+    expect(endGame.text()).toBe("Cat's Game!");
+  });
+  it("can determine when the game is over", () => {
```
Comment (d48) [inline src/test/App.test.js]: It's ok to remove this test

File: src/test/App.test.js [hunk 9]
```diff
@@ -36,9 +41,68 @@ describe("App", () => {
...
+    expect(wrapper.find(EndGameDeclaration)).toHaveLength(0);
+    square.at(0).simulate("click"); //O
+    square.at(5).simulate("click"); //X
+    square.at(1).simulate("click"); //O
+    square.at(6).simulate("click"); //X
+    square.at(2).simulate("click"); //0
+    const endGame = wrapper.find(EndGameDeclaration);
+    expect(endGame.text()).toBe("Player O Won!");
+    const newGame = wrapper.find(NewGame);
+    newGame.find("Button").simulate("click");
```
Comment (d48) [inline src/test/App.test.js:107]: You need to assert a new game has started by the Player being X and the board is blank

**Review by d48 — APPROVED** [review]
Merge and fix my comments

**Review by d48 — APPROVED** [review]
Merge it

**Comment by d48** [comment]
Why isn't this merged yet?

**Pull request merged**